core/src/main/scala/org/apache/spark/rdd/OrderedRDDFunctions.scala

```diff
@@ -72,6 +72,8 @@ class OrderedRDDFunctions[K : Ordering : ClassTag,
    * because it can push the sorting down into the shuffle machinery.
    */
   def repartitionAndSortWithinPartitions(partitioner: Partitioner): RDD[(K, V)] = {
+    val keyClass = implicitly[ClassTag[K]].runtimeClass
+    Partitioner.assertHashCodeIsWellBehaved(keyClass)
```

Comment:
Technically, this method may work even though the hash code is not well-behaved, if the partitioner is not a HashPartitioner, right?